src/Controller/CRUDController.php

```diff
@@ -1520,6 +1520,41 @@ protected function preList(Request $request)
         return $this->get('translator')->trans($id, $parameters, $domain, $locale);
     }
 
+    protected function handleXmlHttpRequestErrorResponse(Request $request, FormInterface $form): ?JsonResponse
+    {
+        if (empty(array_intersect(['application/json', '*/*'], $request->getAcceptableContentTypes()))) {
+            @trigger_error('In next major version response will return 406 NOT ACCEPTABLE without `Accept: application/json` or `Accept: */*`', E_USER_DEPRECATED);
```

Comment:
This message should be explicit about the version where the absence of the header was deprecated.

Reply:
```
Foo is not supported since sonata-projet/admin-bundle 3.x and will return 406 in 4.0
```

Reply:
The PR is about making the methodes protected, this line is not changed. Is it not better to do this change in a separate PR to keep the git commit etc relevant?

Reply:
Oh, it's not a new deprecation indeed.
We should find since when it's deprecated and modify the message, and add a NEXT_MAJOR comment.
It can be done in another PR or in another commit.

Reply:
I have change the deprecation message in a new commit.